brushfire-finatra/src/main/scala/com/stripe/brushfire/finatra/BrushfireServer.scala

```diff
@@ -1,12 +1,13 @@
 package com.stripe.brushfire.finatra
 
 import com.stripe.brushfire._
+import com.twitter.algebird.Semigroup
 import com.twitter.finatra._
 import com.twitter.finagle.http._
 import com.twitter.bijection._
 
 class BrushfireServer extends FinatraServer {
-  def score[K, V, T, P](root: String, trees: Iterable[Tree[K, V, T]], voter: Voter[T, P])(fn: ParamMap => Map[K, V]) = {
+  def score[K, V, T: Semigroup, P](root: String, trees: Iterable[Tree[K, V, T]], voter: Voter[T, P])(fn: ParamMap => Map[K, V]) = {
```

Comment:
It doesn't compile without the constraint for me.
On Aug 29, 2015 12:46 PM, "Avi Bryant" <notifications@github.com> wrote:

> In
> brushfire-finatra/src/main/scala/com/stripe/brushfire/finatra/BrushfireServer.scala
> <https://github.com/stripe/brushfire/pull/62#discussion_r38265258>:
>
> >  import com.twitter.finatra._
> >  import com.twitter.finagle.http._
> >  import com.twitter.bijection._
> >
> >  class BrushfireServer extends FinatraServer {
> > -  def score[K, V, T, P](root: String, trees: Iterable[Tree[K, V, T]], voter: Voter[T, P])(fn: ParamMap => Map[K, V]) = {
> > +  def score[K, V, T: Semigroup, P](root: String, trees: Iterable[Tree[K, V, T]], voter: Voter[T, P])(fn: ParamMap => Map[K, V]) = {
>
> why add the semigroup constraint? This seems like a semantic change rather
> than style/performance.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stripe/brushfire/pull/62/files#r38265258>.
>